src/glm_parser.py

```diff
@@ -255,172 +241,150 @@ def f_argmax(self, sentence):
             "./weight_dump_Iter_1.db",
             "./weight_dump_Iter_2.db"...
             """)
-        arg_parser.add_argument('--frequency', '-f', type=int,
+        arg_parser.add_argument('--dump-frequency', '-f', type=int,
             help="""Frequency of dumping weight vector. The weight vector of last
             iteration will always be dumped
             example: "-i 6 -f 2"
             weight vector will be dumped at iteration 0, 2, 4, 5.
             """)
-        arg_parser.add_argument('--iteration', '-i', metavar='ITERATIONS', type=int,
+        arg_parser.add_argument('--iterations', '-i', metavar='ITERATIONS', type=int,
             help="""Number of iterations
             default 1
             """)
-        arg_parser.add_argument('--timer', '-a', action='store_true',
+        arg_parser.add_argument('--timing', '-a', action='store_true',
             help="""turn on the timer (output time usage for each sentence)
             If combined with --debug-run-number then before termination it also
             prints out average time usage
             """)
-        arg_parser.add_argument('--hadoop', '-c', action='store_true')
-        args=arg_parser.parse_args()
-    #     Initialise sparkcontext
-    sc = None
-    if args.hadoop:
-        h_flag = True
-    if args.spark:
-        parallel_flag = True
-
-    if parallel_flag or h_flag:
-        from pyspark import SparkContext,SparkConf
+        arg_parser.add_argument('--hadoop', '-c', action='store_true',
+           help="""Using Glm Parser in Spark Yarn Mode""")
+
+        args = arg_parser.parse_args()
+
+    # Initialise sparkcontext
+    sparkContext = None
+    yarn_mode  = True if args.hadoop       else False
+    spark_mode = True if args.spark_shards else False
+
+    if spark_mode or yarn_mode:
+        from pyspark import SparkContext, SparkConf
         conf = SparkConf()
-        sc = SparkContext(conf=conf)
+        sparkContext = SparkContext(conf=conf)
 
-    if args.config: # Process config
-        if (not os.path.isfile(args.config)) and (not args.hadoop):
-            raise ValueError('Specified config file does not exist or is not a file: ' + args.config)
+    # Process debug options
+    if args.max_sentences:
+        debug.debug.run_first_num = int(args.max_sentences)
+        if debug.debug.run_first_num <= 0:
+            raise ValueError("Illegal integer: %d" % (debug.debug.run_first_num, ))
+        else:
+            print("Debug run number = %d" % (debug.debug.run_first_num, ))
+    if args.log_feature_request:
+        debug.debug.log_feature_request_flag = True
+        print("Enable feature request log")
+    if args.timing:
+        debug.debug.time_accounting_flag = True
+
+    # Process config
+    if args.config:
+        # Check local config path
+        if (not os.path.isfile(args.config)) and (not yarn_mode):
+            raise ValueError("The config file doesn't exist: " + args.config)
```

Comment:
Since this is within a `if __name__ == '__main__'` there's no way to catch this exception. It can just be a print to `stderr`.